src/client/Client.cc

```diff
@@ -7571,7 +7576,7 @@ int Client::open(const char *relpath, int flags, mode_t mode, int stripe_unit,
     string dname = dirpath.last_dentry();
     dirpath.pop_dentry();
     InodeRef dir;
-    r = path_walk(dirpath, &dir, true, uid, gid);
+    r = path_walk(dirpath, &dir, true, 0, uid, gid);
```

Comment:
We probably ought to request CEPH_CAP_AUTH_SHARED here if client_permissions is true. I'll plan to fold that into this patch once others have had a chance to comment.